libraries/standard/serializer/include/iot_serializer.h

```diff
@@ -275,12 +289,14 @@ typedef struct IotSerializerEncoderObject
 typedef struct IotSerializerDecoderObject
 {
     IotSerializerDataType_t type;
+    const uint8_t * bufferPtr;
+    size_t dataLengthInBuffer;
     union
     {
-        /* useful when this is a container */
+        /* Useful if the type is a container. */
         void * pHandle;
         /* if the type is a container, the scalarValue is unuseful */
-        IotSerializerScalarValue_t value;
+        IotSerializerScalarValue_t scalarValue;
```

Comment:
This may break existing code. Be careful.

Reply:
Good catch, I will revert that name change